src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs

```diff
@@ -449,6 +449,49 @@
    (fn [{:keys [refs this]}]
      (.addEventListener (@refs "filter-field") "search" #(when (= (.-value (.-currentTarget %)) "") (react/call :apply-filter this))))})
 
+(react/defc AutocompleteFilter
+  {:apply-filter
+   (fn [{:keys [props refs]}]
+     ((:on-filter props) (common/get-text refs "autocomplete-filter-field")))
+   :component-did-mount
+   (fn [{:keys [refs props locals this]}]
+     (.addEventListener (@refs "autocomplete-filter-field") "search" #(when (= (.-value (.-currentTarget %)) "") (react/call :apply-filter this)))
+     (let [options (js/Bloodhound.
+                     (clj->js
+                       {:datumTokenizer js/Bloodhound.tokenizers.whitespace
+                        :queryTokenizer js/Bloodhound.tokenizers.whitespace
+                        :remote (clj->js (:bloodhoundInfo props))}))]
+       (.typeahead (js/$ (@refs "autocomplete-filter-field"))
+                   (clj->js
+                     {:hint true
+                      :minLength 1})
+                   (clj->js
+                     {:source options
+                      :display (:typeaheadDisplay props)
+                      :templates
+                      (clj->js
+                        {:empty "<div style='font-size:small; padding: 0.5em;'> Unable to find any matches to the current query </div>"
+                         :suggestion (:typeaheadSuggestionTemplate props)})}))
+       (swap! locals assoc :bloodhoundInstance options))
+     (.bind (js/$ (@refs "autocomplete-filter-field"))
+            "typeahead:select"
+            (fn [ev suggestion]
+              ((:on-filter props) ((:typeaheadDisplay props) suggestion)))))
+   :component-will-receive-props
+   (fn [{:keys [locals]}]
+     (let [bi (:bloodhoundInstance @locals)]
+       (.initialize bi true)))
+   :render
```

Comment:
Lifecycle methods should be in lifecycle order -- `:render` should go above `:component-did-mount`